qiskit/providers/models/pulsedefaults.py

```diff
@@ -144,7 +144,15 @@ def __init__(self,
         self.converter = QobjToInstructionConverter(pulse_library)
         for inst in cmd_def:
             pulse_insts = [self.converter(inst) for inst in inst.sequence]
-            schedule = ParameterizedSchedule(*pulse_insts, name=inst.name)
+            # there is mismatch among instruction sequence and parameter ordering
+            # in u2 and u3 gates.
+            parameters = []
+            if inst.name in ['u2', 'u3']:
```

Comment:
Why don't we just sort the parameters from all cmd_def entries instead of hardcoding here and just define that for now, their ordering is alphabetical?

Reply:
because cmd_def doesn't have parameter. we collect parameter names by making a list of `ParametrizedSchedule`s from `sequence`.
```
{'sequence': [{'t0': 0, 'ch': 'd0', 'name': 'fc', 'phase': '-(P1)'},
  {'t0': 0, 'ch': 'u1', 'name': 'fc', 'phase': '-(P1)'},
  {'t0': 0, 'ch': 'd0', 'name': 'Y90p_d0_076c'},
  {'t0': 160, 'ch': 'd0', 'name': 'fc', 'phase': '-(P0)'},
  {'t0': 160, 'ch': 'u1', 'name': 'fc', 'phase': '-(P0)'}],
 'name': 'u2',
 'qubits': [0]}
```
We can sort `parameters` of `ParametrizedSchedule` of above, but we need to add a setter for this variable. But this is not good solution because schedule objects are designed to be immutable.

Reply:
What I'm saying is why do we need to treat ['u2', 'u3'] as special, why can't we sort parameters for all entries here regardless of their name?

Reply:
I think u2 and u3 are special case. Other provider may define command sequence like
```
 {'t0': 0, 'ch': 'd0', 'name': 'fc', 'phase': 'theta'},
 {'t0': 10, 'ch': 'd0', 'name': 'fc', 'phase': 'phi'},
```
and want to keep the order of parameter so that *args to be `(theta, phi)`. If we sort parameters of all instructions, *args of above becomes `(phi, theta)`. This behavior is not intended by the provider. IMHO, parameter order mismatch of u2 and u3 is kind of convention but not usual.

Reply:
I disagree, I think we must assume alphabetical ordering for commands coming from the backend cmd_def in JSON format since there is no way to specify argument ordering as a parameter header is missing